pytest_plugins/metadata_markers.py

```diff
@@ -56,6 +90,57 @@ def pytest_configure(config):
     re.IGNORECASE,
 )
 
+blocked_by_regex = re.compile(
+    # To match :BlockedBy: SAT-32932
+    r'\s*:BlockedBy:\s*(?P<blocked_by>.*\S*)',
+    re.IGNORECASE,
+)
+
+verifies_regex = re.compile(
+    # To match :Verifies: SAT-32932
+    r'\s*:Verifies:\s*(?P<verifies>.*\S*)',
+    re.IGNORECASE,
+)
+
+
+def handle_verification_issues(item, verifies_marker, verifies_issues):
+    """Handles the logic for deselecting tests based on Verifies testimony token
+    and --verifies-issues pytest option.
+    """
+    if verifies_issues:
+        if not verifies_marker:
+            log_and_deselect(item, '--verifies-issues')
+            return False
+        if isinstance(verifies_issues, list):
+            verifies_args = verifies_marker.args[0]
+            if all(issue not in verifies_issues for issue in verifies_args):
+                log_and_deselect(item, '--verifies-issues')
+            return False
+    return True
```

Comment:
may be  reduce the nesting of if statements

Reply:
Tried my best with that already😅 did quite brainstorming to get to this while keeping all the scenarios in mind😄